include/dlaf/mc/triangular_solve_local.h

```diff
@@ -0,0 +1,280 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+#include <hpx/include/parallel_executors.hpp>
+#include <hpx/include/threads.hpp>
+
+#include "dlaf/blas_tile.h"
+#include "dlaf/common/index2d.h"
+#include "dlaf/lapack_tile.h"
+#include "dlaf/matrix.h"
+#include "dlaf/matrix/distribution.h"
+#include "dlaf/util_matrix.h"
+
+/// @file
+
+namespace dlaf {
+/// @brief Triangular Solve implementation on local memory, solving op(A) X = alpha B (when side == Left)
+/// or X op(A) = alpha B (when side == Right).
+///
+/// @param side specifies whether op(A) appears on the \a Left or on the \a Right of dlaf::Matrix X
+/// @param uplo specifies whether the dlaf::Matrix A is a \a Lower or \a Upper triangular matrix
+/// @param op specifies the form of op(A) to be used in the matrix multiplication: \a NoTrans, \a Trans,
+/// \a ConjTrans
+/// @param diag specifies if the dlaf::Matrix A is assumed to be unit triangular (\a Unit) or not (\a NonUnit)
+/// @param alpha specifies the scalar alpha
+/// @tparam A refers to a triangular dlaf::Matrix object
+/// @tparam B refers to a dlaf::Matrix object
+template <class T>
+void triangular_solve(blas::Side side, blas::Uplo uplo, blas::Op op, blas::Diag diag, T alpha,
+                      Matrix<T, Device::CPU>& A, Matrix<T, Device::CPU>& B) {
```

Comment:
The matrix `a` should be read-only, therefore its type should be `Matrix<const T, Device::CPU>&`